Scripts/update-translations.rb

```diff
@@ -124,27 +119,44 @@
   langs = ALL_LANGS
 end
 
+script_root = __dir__
+project_dir = File.dirname(script_root)
+
 langs.each do |code,local|
-  lang_dir = File.join('WordPress', 'Resources', "#{local}.lproj")
-  puts "Updating #{code}"
+  lang_dir = File.join(project_dir, 'WordPress', 'Resources', "#{local}.lproj")
+  puts "Updating #{code} in #{lang_dir}"
   system "mkdir -p #{lang_dir}"
 
-  # Download for production
-  system "if [ -e #{lang_dir}/Localizable#{strings_file_ext}.strings ]; then cp #{lang_dir}/Localizable#{strings_file_ext}.strings #{lang_dir}/Localizable#{strings_file_ext}.strings.bak; fi"
+  destination = "#{lang_dir}/Localizable#{strings_file_ext}.strings"
+  backup_destination = "#{destination}.bak"
+
+  # Step 2 – Download the new strings
+  if File.exist? destination
+    FileUtils.copy destination, backup_destination
+  end
 
   url = "#{download_url}/#{code}/default/export-translations?#{strings_filter}format=strings"
-  destination = "#{lang_dir}/Localizable#{strings_file_ext}.strings"
 
-  system "curl -fLso #{destination} \"#{url}\"" or begin
+  system "curl -fgLo #{destination} \"#{url}\"" or begin
     puts "Error downloading #{code}"
   end
 
-  if File.size(destination).to_f == 0
-    abort("\e[31mFatal Error: #{destination} appears to be empty. Exiting.\e[0m")
+  # Step 3 – Validate the new file
+  if !File.exist?(destination) or File.size(destination).to_f == 0
+    puts "\e[31mFatal Error: #{destination} appears to be empty. Exiting.\e[0m"
+    abort()
   end
 
-  system "./Scripts/fix-translation #{lang_dir}/Localizable#{strings_file_ext}.strings"
-  system "plutil -lint #{lang_dir}/Localizable#{strings_file_ext}.strings" and system "rm #{lang_dir}/Localizable#{strings_file_ext}.strings.bak"
+  fix_script_path = File.join(script_root, 'fix-translation')
+
+  system "#{fix_script_path} #{lang_dir}/Localizable#{strings_file_ext}.strings"
+  system "plutil -lint #{lang_dir}/Localizable#{strings_file_ext}.strings"
```

Comment:
iinm, `system`, like many similar methods dealing with executing shell commands, has a safer version which accepts taking each shell parameter as separate method parameters. Which is safer because you don't need to be sure not to forget to shell-escape everything or put each parameter inside quotes manually.

```ruby
# unsafe version: will fail if e.g. either `fix_script_path` or `lang_dir` happens to contain spaces
system "#{fix_script_path} #{lang_dir}/Localizable#{strings_file_ext}.strings"
# manual, safer version: quoting everything manually. Could fail if one of the variable expansion itself contains quotes, and still makes it open to injection and security vulnerabilities
system "\"#{fix_script_path}\" \"#{lang_dir}/Localizable#{strings_file_ext}.strings\""
# Safer way, letting system make everything secure and properly escaped (actually, avoids involving the shell interpreter internally to let it split the string into parameters, and instead popen the execution process with the right argvs directly)
# and thus doesn't risk any issue or security vulnerability with spaces, unescaped-quotes, or env variable injection
system(fix_script_path, "#{lang_dir}/Localizable#{strings_file_ext}.strings")
```

(only commenting there, but ofc apply same for every other call to `system`)

Reply:
Oh nice – I like this quite a lot – will run around and audit for it

Reply:
Applied in bb46c203005ff89ffdeadf292ebe194207943663